### PR TITLE
feat: reference section updates

### PR DESCRIPTION
**Pull request by 0xPenryn**
updates reference section of documentation for idkit v1

### CODE REVIEW

**Review by m1guelpf — APPROVED** [review]

File: src/pages/reference/api.mdx [hunk 1]
```diff
@@ -133,21 +133,55 @@ fetch(apiUrl, {
...
+		The action to check. This is the same action that was passed to IDKit. Ignored if an `external_nullifier` is provided. Defaults to an empty string for Sign In with Worldcoin.
 	</Property>
-	<Property name="nullifier_hash" type="string">
-		The ZKP's nullifier hash, as provided by IDKit. See [IDKit response](/idkit/reference#response) for details.
+	<Property name="nullifier_hash" type="string" defaultValue='""'>
+		The ZKP's nullifier hash, as provided by IDKit. See [IDKit response](/idkit/reference#response) for details. Defaults to an empty string.
 	</Property>
-	<Property name="external_nullifier" type="string">
-		The ZKP's external nullifier.
+	<Property name="external_nullifier" type="string" defaultValue='null'>
```
Comment (pdtfh) [inline src/pages/reference/api.mdx:151]: Let's already mark external_nullifier as deprecated. Use in Dev Portal should be discouraged and removed soon. The wallets will now handle external nullifier generation.

File: src/pages/reference/api.mdx [hunk 2]
```diff
@@ -133,21 +133,55 @@ fetch(apiUrl, {
...
+	<Property name="is_verified" type="string">
+		Whether the app has been verified by Worldcoin.
+	</Property>
+	<Property name="name" type="string">
+		The app's name as configured in the Developer Portal.
+	</Property>
+	<Property name="verified_app_logo" type="string">
+		The URL of the app's logo. Only returned if the app has been verified, otherwise returns an empty string.
+	</Property>
+	<Property name="sign_in_with_world_id" type="boolean">
```
Comment (pdtfh) [inline src/pages/reference/api.mdx:177]: We're renaming this to `is_sign_in`

File: src/pages/reference/api.mdx [hunk 3]
```diff
@@ -133,21 +133,55 @@ fetch(apiUrl, {
...
+		Whether the action request is for Sign In with Worldcoin.
+	</Property>
+	<Property name="can_user_verify" type='"yes" | "no" | "on-chain" | "undetermined"'>
+		If the user is eligible to verify for this action. Will always return `"yes"` if `sign_in_with_world_id` is `true`, or `"on-chain"` for **On-Chain** apps. For **Cloud** apps, will return `"yes"` or `"no"` if a `nullifier_hash` was provided, or `"undetermined"` if a `nullifier_hash` was not provided.
```
Comment (pdtfh) [inline src/pages/reference/api.mdx:181]: let's make this clearer, particularly for yes & no it depends on not only if the nullifier hash is passed but also if the user has verified for this action before and the max number of verifications for this action. let's explain why we can validate this for cloud only.

File: src/pages/reference/api.mdx [hunk 4]
```diff
@@ -133,21 +133,55 @@ fetch(apiUrl, {
...
+	</Property>
+	<Property name="action" type="JSON">
```
Comment (pdtfh) [inline src/pages/reference/api.mdx:183]: let's add details on this object,

```
interface _Action
  extends Pick<
    ActionModel,
    | "name"
    | "description"
    | "max_verifications"
    | "max_accounts_per_user"
    | "action"
    | "external_nullifier"
    | "status"
    | "__typename"
  > {
}
```

File: src/pages/reference/api.mdx [hunk 5]
```diff
@@ -206,6 +239,30 @@ fetch(`/api/v1/precheck/${appId}`, {
...
+{
+	"code": "required",
+	"detail": "This attribute is required for new actions.",
+	"attribute": "action"
+}
+```
+
+```json {{ title: "400 External Nullifier Mismatch" }}
+{
+	"code": "external_nullifier_mismatch",
```
Comment (pdtfh) [inline src/pages/reference/api.mdx:252]: remove

File: src/pages/reference/idkit.mdx [hunk 6]
```diff
@@ -5,91 +5,88 @@
...
-</Properties>
-
-### Optional
-
-<Properties>
-	<Property name="handleVerify" type="function">
-		Called after the proof is returned from the World App, but before showing the success screen. Throwing an error in this
-		screen will show the user a custom error. Used to perform additional validation when needed.
+	<Property name="handleVerify" type="function(ISuccessResult)">
+		Called after the proof is returned from the World App, but before showing the success screen. Should receive a single parameter of type `ISuccessResult`
```
Comment (pdtfh) [inline src/pages/reference/idkit.mdx:51]:
```suggestion
		Called after the proof is returned from the user's identity wallet (e.g. World App), but before showing the success screen. Should receive a single parameter of type `ISuccessResult`
```

File: src/pages/reference/idkit.mdx [hunk 7]
```diff
@@ -5,91 +5,88 @@
...
+		which contains [the proof details.](#response) Throwing an error in this screen will show the user a custom error. Used to perform additional validation when needed.
```
Comment (pdtfh) [inline src/pages/reference/idkit.mdx:52]: let's add examples of when this is useful. for most cases this also provides better UX, let's encourage use.

File: src/pages/reference/idkit.mdx [hunk 8]
```diff
@@ -189,33 +166,60 @@ The JS package may return any of the following error codes.
...
+			<td>The user might have the requested credential, but it is not available on-chain yet. It might be available for API verification.</td>
+			<td>Ask the user to verify again in approximately one hour.</td>
+		</tr>
+		<tr>
+			<td>`invalid_network`</td>
+			<td>The application is configured for a different environment than the verifying user's client.</td>
+			<td>Ensure you use the [Worldcoin Simulator](https://simulator.worldcoin.org) for Staging applications and World App for Production applications.</td>
+		</tr>
+		<tr>
+			<td>`metadata_failed`</td>
```
Comment (pdtfh) [inline src/pages/reference/idkit.mdx:220]: review updated error codes, I believe some are missing

File: src/pages/reference/idkit.mdx [hunk 9]
```diff
@@ -236,33 +240,87 @@ The JS package may return any of the following error codes.
...
 
-## Telemetry
+## IDKit Standalone
+
+The `@worldcoin/idkit-standalone` package is intended for vanilla JS applications. It is a standalone package that acts as a wrapper around the `@worldcoin/idkit` package.
+
+### Methods
+
+<Note>
+	The `.init()` and `.update()` methods take the same parameters as the React package's IDKitWidget component. See [above](#id-kit-react) for more details.
```
Comment (pdtfh) [inline src/pages/reference/idkit.mdx:250]: These are no longer documented above

File: src/pages/reference/idkit.mdx (hunk 8 above)
Comment (0xPenryn) [reply src/pages/reference/idkit.mdx:220]: i included all error codes listed in the world id 2.0 and bridge specs, except those that are not returned to IDKit

File: src/pages/reference/idkit.mdx (hunk 9 above)
Comment (0xPenryn) [reply src/pages/reference/idkit.mdx]: fixed link to go directly to parameters

File: src/pages/reference/api.mdx [hunk 10]
```diff
@@ -8,34 +8,34 @@ The Worldcoin Developer Portal offers an API to enable easy verification of Worl
 https://developer.worldcoin.org
 ```
 
-## Verify Proof {{ tag: "POST", label: "/v1/verify/{app_id}" }}
```
[the remark this answers is not in the document]
Comment (0xPenryn) [reply src/pages/reference/api.mdx:11]: will be included in a separate PR

**Pull request merged**